tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py

```diff
@@ -0,0 +1,149 @@
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_delayed(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs)
+        P.set_yi(ys)
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_append(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs[:3], ys[:3])
+        P.add_xi(xs[3:], ys[3:])
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_vector(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        xs = xp.array([0, 1, 2], dtype=dtype)
+        ys = xp.array([[0, 1], [1, 0], [2, 1]], dtype=dtype)
+        test_xs = xp.linspace(-1, 3, 100, dtype=dtype)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(test_xs)
+
+    @pytest.mark.parametrize('test_xs', [0, [0], [0, 1]])
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_shapes_scalarvalue(self, xp, scp, test_xs):
+        true_poly = xp.poly1d([-2, 3, 5, 1, -3])
+        xs = xp.linspace(-1, 10, 10)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        test_xs = xp.array(test_xs)
+        return xp.shape(P(test_xs))
+
+    @pytest.mark.parametrize('test_xs', [0, [0], [0, 1]])
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_shapes_vectorvalue(self, xp, scp, test_xs):
+        true_poly = xp.poly1d([4, -5, 3, 2, -4])
+        xs = xp.linspace(-10, 10, 20)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(
+            xs, xp.outer(ys, xp.arange(3)))
+        test_xs = xp.array(test_xs)
+        return xp.shape(P(test_xs))
+
+    @pytest.mark.parametrize('test_xs', [0, [0], [0, 1]])
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_shapes_1d_vectorvalue(self, xp, scp, test_xs):
+        true_poly = xp.poly1d([-3, -1, 4, 9, 8])
+        xs = xp.linspace(-1, 10, 10)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(
+            xs, xp.outer(ys, xp.array([1])))
+        test_xs = xp.array(test_xs)
+        return xp.shape(P(test_xs))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_large_chebyshev(self, xp, scp, dtype):
```

Comment:
The CI failure in test_large_chebyshev is because of scipy's bug fixed at https://github.com/scipy/scipy/pull/14255. We can only test it with scipy>=1.8.0.

```suggestion
    @testing.with_requires("scipy>=1.8.0")
    def test_large_chebyshev(self, xp, scp, dtype):
```

Reply:
Thanks!